### PR TITLE
added support to iterate over multiple images with the same title

### PR DESCRIPTION
**Pull request by juspky**
the added **allowMultiple: true** now allows:
`
sc.Entity.Images.ForEach(i => ReplaceImage(wordDoc, i, "Image", $"imagePart{i.Suffix}"));`

one

This is very helpfull when you want to list all images of an MList in a report

### CODE REVIEW

File: Signum.Engine.Extensions/Word/WordImageReplacer.cs [hunk 1]
```diff
@@ -82,13 +83,19 @@ private static ImageFormat ToImageFormat(ImagePartType imagePartType)
...
         {
-            var drawing = doc.MainDocumentPart.Document.Descendants().OfType<Drawing>().SingleEx(r =>
+            var drawing = doc.MainDocumentPart.Document.Descendants().OfType<Drawing>().First(r =>
             {
                 var prop = r.Descendants<DocProperties>().SingleOrDefault();
+                var match = prop != null && (prop.Title == titleOrDescription || prop.Description == titleOrDescription);
+
+                if (allowMultiple && match)
+                {
+                    prop.Title += Guid.NewGuid().ToString();
```
Comment (olmobrutall) [inline Signum.Engine.Extensions/Word/WordImageReplacer.cs]: What does line does?

Comment (juspky) [reply Signum.Engine.Extensions/Word/WordImageReplacer.cs]: It changes the DocProperty xml tag to a different one so it only gets matched once

**Comment by olmobrutall** [comment]
But then you don’t delete the original bilp at all. Looks hacky. Instead make a new method ReplaceAllImages that asserts that there are N images in the document and you take N images as parameter.

**Comment by juspky** [comment]
> 
> 
> But then you don’t delete the original bilp at all. Looks hacky. Instead make a new method ReplaceAllImages that asserts that there are N images in the document and you take N images as parameter.

Done

**Comment by juspky** [comment]
@olmobrutall ping

File: Signum.Engine.Extensions/Word/WordImageReplacer.cs [hunk 2]
```diff
@@ -84,15 +119,32 @@ private static ImageFormat ToImageFormat(ImagePartType imagePartType)
...
         }
+
+        static Blip[] FindAllBlips(WordprocessingDocument doc, string titleOrDescription)
+        {
+            var i = 0;
+            var drawing = doc.MainDocumentPart.Document.Descendants().OfType<Drawing>().Where(r =>
+            {
+                var prop = r.Descendants<DocProperties>().SingleOrDefault();
+                var match = prop != null && (prop.Title == titleOrDescription || prop.Description == titleOrDescription);
+                prop.Title += i;
```
Comment (olmobrutall) [inline Signum.Engine.Extensions/Word/WordImageReplacer.cs]: Why you need to change the title?

File: Signum.Engine.Extensions/Word/WordImageReplacer.cs [hunk 3]
```diff
@@ -18,7 +19,7 @@ public class WordImageReplacer
         public static bool AvoidAdaptSize = false; //Jpeg compression creates different images in TeamCity
 
         /// <param name="titleOrDescription">Word Image -> Right Click -> Format Picture -> Alt Text -> Title </param>
-        public static void ReplaceImage(WordprocessingDocument doc, string titleOrDescription, Bitmap bitmap, string newImagePartId, bool adaptSize = false, ImagePartType imagePartType = ImagePartType.Png)
+        public static void ReplaceImage(WordprocessingDocument doc, string titleOrDescription, Bitmap bitmap, string newImagePartId, bool adaptSize = false, bool allowMultiple = false, ImagePartType imagePartType = ImagePartType.Png)
```
Comment (olmobrutall) [inline Signum.Engine.Extensions/Word/WordImageReplacer.cs]: AllowMultiple is not necessary anymore

File: Signum.Engine.Extensions/Word/WordImageReplacer.cs [hunk 4]
```diff
@@ -84,15 +119,32 @@ private static ImageFormat ToImageFormat(ImagePartType imagePartType)
 
         static Blip FindBlip(WordprocessingDocument doc, string titleOrDescription)
         {
-            var drawing = doc.MainDocumentPart.Document.Descendants().OfType<Drawing>().SingleEx(r =>
+            var drawing = doc.MainDocumentPart.Document.Descendants().OfType<Drawing>().Single(r =>
```
Comment (olmobrutall) [inline Signum.Engine.Extensions/Word/WordImageReplacer.cs]: SingleEx throws a better exception

**Comment by olmobrutall** [comment]
I’ve made some suggestions, but I merge the PR anyway! Thanks :D

**Pull request merged**